modular_skyrat/modules/automapper/code/area_spawn_entries.dm

```diff
@@ -0,0 +1,159 @@
+					density_found = TRUE
+					break
+			if(density_found)
+				continue
+			// Time to check cardinals
+			// The cardinals must be clear of any density, aside from wall cardinals!
+			var/cardinal_density_check = TRUE
+			// Some items want to be hugging a wall, such as lockers, and machines. It just looks better.
+			var/wall_check = FALSE
+			for(var/dir in GLOB.cardinals)
+				var/turf/cardinal_test_turf = get_step(iterating_turf, dir)
+				if(isclosedturf(cardinal_test_turf))
+					wall_check = TRUE
+				for(var/atom/movable/found_movable in cardinal_test_turf)
+					if(found_movable.density && !wall_check)
+						cardinal_density_check = FALSE
+						break
+			if(wall_hug && !wall_check)
+				continue
+			if(!cardinal_density_check && wall_hug)
+				continue
+			// Finally we want to prioritise entirely empty turfs
+			var/totally_empty = TRUE
+			for(var/atom/movable/found_movable in iterating_turf)
+				totally_empty = FALSE
+				break
+			if(totally_empty)
+				available_empty_turfs += iterating_turf
+			available_turfs += iterating_turf
+
+	if(!LAZYLEN(available_turfs))
+		CRASH("[src.type] could not find any suitable turfs on map [SSmapping.config.map_name]!")
+
+	for(var/i in 1 to amount_to_spawn)
+		if(LAZYLEN(available_empty_turfs))
+			new desired_atom(pick(available_empty_turfs))
+		else
+			new desired_atom(pick(available_turfs))
+
+// Pets
+/datum/area_spawn/markus
+	target_areas = list(/area/station/cargo/sorting,  /area/station/cargo/storage, /area/station/cargo/office, /area/station/cargo/qm)
+	desired_atom = /mob/living/simple_animal/pet/dog/markus
+
+/datum/area_spawn/bumbles
+	target_areas = list(/area/station/service/hydroponics, /area/station/service/hydroponics/upper)
+	desired_atom = /mob/living/simple_animal/pet/bumbles
+
+/datum/area_spawn/borgi
+	target_areas = list(/area/station/science/robotics, /area/station/science/robotics/mechbay, /area/station/science/robotics/lab)
+	desired_atom = /mob/living/simple_animal/pet/dog/corgi/borgi
+
+/datum/area_spawn/poppy
+	target_areas = list(/area/station/engineering/main, /area/station/engineering/break_room, /area/station/engineering/lobby, /area/station/engineering/supermatter/room)
+	desired_atom = /mob/living/simple_animal/pet/poppy
+
+// Structures
+/datum/area_spawn/secmed_locker
+	target_areas = list(/area/station/security/medical)
+	desired_atom = /obj/structure/closet/secure_closet/security_medic
+	wall_hug = TRUE
+
+/datum/area_spawn/command_drobe
+	target_areas = list(/area/station/command/bridge, /area/station/command/meeting_room)
+	desired_atom = /obj/machinery/vending/access/command
+	wall_hug = TRUE
+
+/datum/area_spawn/lustwish_dorms
+	target_areas = list(/area/station/commons/locker, /area/station/commons/dorms)
+	desired_atom = /obj/machinery/vending/dorms
+	wall_hug = TRUE
+
+/datum/area_spawn/lustwish_prison
+	target_areas = list(/area/station/security/prison, /area/station/security/prison/shower)
+	desired_atom = /obj/machinery/vending/dorms
```

Comment:
Didn't we have a special version of it that was unique to the prison, which didn't have tools in it?

Reply:
cant find one